client/lib/cart-values/cart-items.js

```diff
@@ -647,6 +653,22 @@ function getIncludedDomain( cartItem ) {
 	return cartItem.extra && cartItem.extra.includedDomain;
 }
 
+function setVolume( cartItem, volume ) {
+	function setItemVolume( items ) {
+		return items.map( item => {
+			if ( item.meta === cartItem.meta && ( item.product_id === cartItem.product_id || isPrivacyProduct( item ) ) ) {
+				return extend( {}, item, { volume } );
+			}
+
+			return item;
+		} );
+	}
+
+	return function( cart ) {
+		return React.addons.update( cart, { products: { $apply: setItemVolume } } );
+	};
+}
```

Comment:
since we are already defining some helpers inside this function, why not add a few more?

```js
function setVolume( item, volume ) {
	const matchesItem = i => (
		i.meta === item.meta &&
		( i.product_id === item.product_id || isPrivacyProduce( item ) )
	);

	const extendIfItemMatches = i => ! matchesItem( i ) ? i : { ...item, volume };
	
	return cart => { ...cart, products: cart.products.map( extendIfItemMatches ) };
}
```

also in this code I didn't use the React immutability helper because it didn't seem to add much here other than make it more confusing for me to read. we now have some great operators in JavaScript that do just what we want for that. It would actually still be more useful if `products` were some deeply-nested property inside of `cart` but that just isn't the case here.